community/kernel/src/main/java/org/neo4j/kernel/impl/store/NeoStore.java

```diff
@@ -1020,4 +1041,35 @@ public boolean visit( PageCursor element ) throws IOException
             }
         } );
     }
+
+    private static class HighestCommittedTransaction
```

Comment:
Actually scratch that. Why don't we make a proper class/interface out of `HighestCommittedTransaction` and have such returned from `getLastCommittedTransaction` instead. This would remove the `long[]` allocation on get _and_ make the returned object more understandable, also immutable. Win-win!